docs/_dapp/abigen.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Abigen
+sort-key: p
+---
+
+Abigen is a binding-generator for easily interacting with Ethereum using Go. 
+Abigen creates easy-to-use, type-safe Go packages from Ethereum smart contract 
+definitions known as ABIs. This abstracts away a lot of the complexity of handling 
+smart contract deployment and interaction in Go native applications such as 
+encoding and decoding events. Abigen comes bundled with 
+Geth. Abigen can also be built 
+independently by navigating to `go-ethereum/cmd/abigen` and running `go build`, or 
+equivalently:
+
+```sh
+$ cd $GOPATH/src/github.com/ethereum/go-ethereum
+$ go build ./cmd/abigen
+```
+
+## What is an ABI?
+
+Ethereum smart contracts have a schema that defines its functions and return types 
+in the form of a JSON file. This JSON file is known as an _Application Binary Interface_, 
+or ABI. The ABI acts as a specification for precisely how to encode data sent to a 
+contract and how to decode the data the contract sends back. The ABI is the only 
+essential piece of information required to generate Go bindings. Go developers can then 
+use the bindings to interact with the contract from their Go application without having 
+to deal directly with data encoding and decoding. An ABI is generated when a contract 
+is compiled.
+
+### Generating the bindings
+
+To demonstrate the binding generator a contract is required. The contract `Storage.sol` 
+implements two very simple functions: `store` updates a user-defined `uint256` to the 
```

Comment:
perhaps `user-provided`?

`uint256` is builtin type, feels confused about `user-defined`

Reply:
```suggestion
implements two very simple functions: `store` updates a user-provided `uint256` to the 
```